### PR TITLE
Test Fix: various API Management tests

### PR DESCRIPTION
**Pull request by mbfrahry**

### CODE REVIEW

File: azurerm/internal/services/apimanagement/api_management_backend_resource_test.go [hunk 1]
```diff
@@ -52,10 +52,10 @@ func TestAccApiManagementBackend_allProperties(t *testing.T) {
...
 				check.That(data.ResourceName).Key("credentials.0.authorization.0.scheme").HasValue("scheme"),
 				check.That(data.ResourceName).Key("credentials.0.certificate.0").Exists(),
-				check.That(data.ResourceName).Key("credentials.0.header.header1").HasValue("header1value1"),
-				check.That(data.ResourceName).Key("credentials.0.header.header2").HasValue("header2value1"),
-				check.That(data.ResourceName).Key("credentials.0.query.query1").HasValue("query1value1"),
-				check.That(data.ResourceName).Key("credentials.0.query.query2").HasValue("query2value1"),
+				check.That(data.ResourceName).Key("credentials.0.header.header1").HasValue("header1,value1"),
+				check.That(data.ResourceName).Key("credentials.0.header.header2").HasValue("header2,value1"),
+				check.That(data.ResourceName).Key("credentials.0.query.query1").HasValue("query1,value1"),
+				check.That(data.ResourceName).Key("credentials.0.query.query2").HasValue("query2,value1"),
```
Comment (jackofallops) [inline azurerm/internal/services/apimanagement/api_management_backend_resource_test.go:58]: Looks like the config has different values here?
```
     header = {
      header1 = "header1value1,header1value2"
      header2 = "header2value1,header2value2"
    }
    query = {
      query1 = "query1value1,query1value2"
      query2 = "query2value1,query2value2"
    }
```

File: azurerm/internal/services/apimanagement/api_management_backend_resource_test.go [hunk 2]
```diff
@@ -326,8 +326,8 @@ resource "azurerm_api_management_backend" "test" {
     ]
     max_partition_resolution_retries = 5
     server_certificate_thumbprints = [
```
Comment (jackofallops) [inline azurerm/internal/services/apimanagement/api_management_backend_resource_test.go:328]: This is a Set?

**Review by jackofallops — APPROVED** [review]
Thanks @mbfrahry - Just a couple observations below, otherwise LGTM 👍

File: azurerm/internal/services/apimanagement/api_management_backend_resource_test.go (hunk 1 above)
Comment (mbfrahry) [reply azurerm/internal/services/apimanagement/api_management_backend_resource_test.go:58]: Ooo, good call. Got it

File: azurerm/internal/services/apimanagement/api_management_backend_resource_test.go (hunk 2 above)
Comment (mbfrahry) [reply azurerm/internal/services/apimanagement/api_management_backend_resource_test.go:328]: It is a set!

**Pull request merged**